Fix for camera not Start, Audio and others

rubencarneiro opened the pull request.

rubencarneiro committed:
Add ubuntu overlay files
Add path to averlay on config

rubencarneiro committed:
add audio info needed by hwaudio

rubencarneiro committed:
Missing speaker configuration

rubencarneiro committed:
Missing vendor libs paths
for audio

rubencarneiro committed:
Fixing error on hwaudio related to TTY
Increase Volume

rubencarneiro committed:
Update description just a cosmetic change

rubencarneiro committed:
Build halium-boot

rubencarneiro committed:
Remove touch symlinks since its already declared on init.hammerhead.rc

rubencarneiro committed:
Clean comments

rubencarneiro committed:
remove duplicate udev rules

rubencarneiro committed:
Remove audioflingerglue

rubencarneiro committed:
Fix autobrightness actually with removing minimediaservice and other …

rubencarneiro committed:
Remove minimedia service from init.rc

rubencarneiro committed:
Add support for 32 bit to abstraction base

rubencarneiro committed:
Add support for car hand free

rubencarneiro committed:
Make a cleaner Android Products

rubencarneiro committed:
Revert changes on audio effectas as dont make any diference to ubuntu touch build.

rubencarneiro committed:
Make a cleaner Products Build

rubencarneiro committed:
clean device tree from assets that we dont need

rubencarneiro committed:
This is neede and requested by Hwaudio to identify sound device.

rubencarneiro committed:
Reverts previous changes as make no effect on ubuntu touch

rubencarneiro committed:
Remove all files that we dont use.

rubencarneiro committed:
Discard changes made we dont want to mess with this file

rubencarneiro committed:
Update the fotmware partition to match the same as lineage builds.

rubencarneiro committed:
Remove the car support

rubencarneiro committed:
Update vendor setup to build lineage instead aosp since building aosp will break the build of libminisf or droidmedia.

rubencarneiro committed:
Update power hall to match lineage os 14.1

rubencarneiro committed:
Update to lineage dependencies

rubencarneiro committed:
Add more permission to wifi display

rubencarneiro committed:
Update gps.conf to lineage 14.1

rubencarneiro committed:
Remove some definitions already defined on lineage.mk

rubencarneiro committed:
Update to remove fp builds we dont need those

rubencarneiro committed:
Updated to lineage os 14.1

rubencarneiro committed:
Remove so fp build that we dont use, also add adaptation for 32 libs only on halium.

rubencarneiro committed:
Add some more halium libs
Disable minimedia fake sensor
remove unused fp builds configs

rubencarneiro committed:
add lineage device build mk

rubencarneiro committed:
Remove this since we are not using car builds configs

rubencarneiro committed:
remove some animation screen to update to lineage os 14.1

rubencarneiro committed:
Update to match lineage os 14.1

rubencarneiro committed:
Update to symlink codecs.xml and firmware

rubencarneiro committed:
Update rules

rubencarneiro committed:
Try to add multilib support

rubencarneiro committed:
Removed for now wont make any change on ubunntu build

rubencarneiro committed:
Add Ubuntu overlay files just the necessary for now

rubencarneiro committed:
Update Leds
Add qmuxd to root
and add fix to cpu online

rubencarneiro committed:
add usb support not enable for now its not fully working

rubencarneiro committed:
This let enable audio calls, but handset is not working

rubencarneiro committed:
Add usb MTP support for halium not fully work

rubencarneiro committed:
camera: Use generated kernel headers

rubencarneiro committed:
Update Readme

rubencarneiro committed:
Update

rubencarneiro committed:
Update

rubencarneiro committed:
Revert changes to headers

rubencarneiro committed:
Enable MTP Supports
add quirks for audio

rubencarneiro committed:
Trying to fix camera
Change mtp group to plugdev to fix MTP

rubencarneiro committed:
Update

rubencarneiro committed:
Update

rubencarneiro committed:
Remove to necessary builds products only

rubencarneiro committed:
remove extras  lines

rubencarneiro committed:
Update a2pdp imput

rubencarneiro committed:
revert some changes to cm builds

rubencarneiro committed:
Change PRIVATE_BUILD_DESC

rubencarneiro committed:
Revert changes on modem may cause lost of imei

rubencarneiro committed:
Revert release tools  SystemSize definition

rubencarneiro committed:
Revert changes to permissions on wifi display

rubencarneiro committed:
Revert changes to gps.conf

rubencarneiro committed:
Revert remove declaration POWER_HINT_VSYNC

rubencarneiro committed:
Donte need these

rubencarneiro committed:
Bluetooth config for car

rubencarneiro committed:
Remove WBS

rubencarneiro committed:
revert changes o position lines por cpu_online

rubencarneiro committed:
Update blbos text

rubencarneiro committed:
Update to cm config

rubencarneiro committed:
Update

rubencarneiro committed:
Update

rubencarneiro committed:
Update Readme

rubencarneiro committed:
update

rubencarneiro committed:
Add minimedia and minisf service
Fixed Camera

rubencarneiro committed:
Fix some compiling issues
Add More ubuntu packages

rubencarneiro committed:
Revert changes dont need to change this makes no diference

rubencarneiro committed:
Change quirks to source.primary for audio in calls
Enable MTP
remove gst-droid it availble only on halium 9

rubencarneiro committed:
Make default permissions on camera 0660
Add gemini0 to rules

rubencarneiro committed:
Update Readme

rubencarneiro committed:
Fix typo

rubencarneiro committed:
Remove some ubuntu overlay files that makes no impact or diference

rubencarneiro committed:
Remove some ubuntu overlay files not needed

rubencarneiro committed:
Remove Duplicate symlink

rubencarneiro committed:
Fix Typo Error

rubencarneiro committed:
Fix Typo Error

rubencarneiro committed:
Remove time_daemon conflits with ubuntu timekeep

rubencarneiro committed:
Merge branch 'halium-7.1-updated' of https://github.com/rubencarneiro/android_device_lge_hammerhead into halium-7.1-updated

rubencarneiro committed:
Fix copy paste error

Flohack74 commented on ubuntu/usb/mtp-server.conf:
```diff
@@ -0,0 +1,27 @@
+description "MTP protocol server"
```
You can remove this, that file is in an identical version already in the root filesystem.

rubencarneiro replied:
removed

Flohack74 commented on device.mk:
```diff
@@ -141,8 +134,30 @@ PRODUCT_CHARACTERISTICS := nosdcard
+    charger_res_images
+
+PRODUCT_PACKAGES += \
+    brcm_patchram_plus \
```
You can remove patchram and hciattach, the patched version is already in the root filesystem.

rubencarneiro replied:
Fixed

Flohack74 commented on device.mk:
```diff
@@ -100,35 +101,27 @@ PRODUCT_COPY_FILES += \
-PRODUCT_PACKAGES += \
-    charger_res_images
+PRODUCT_COPY_FILES += \
+    $(LOCAL_PATH)/ubuntu/usb/setupusb:system/halium/usr/share/usbinit/setupusb \
```
Files in system/halium will be overlaid / bind-mounted into the rootfs. But you are trying to put a file there which does not exist in the rootfs, so you use it more like a generic place for that file. But this could confuse people expecting that system/halium/usr/share/usbinit/ is a valid path in the rootfs. Please put it into /system/bin only, and then adapt the paths in mtp-state.conf see below

Flohack74 commented on ubuntu/usb/mtp-state.conf:
```diff
@@ -0,0 +1,24 @@
+     ;;
+   mtp,adb)
+     /bin/bash /system/halium/usr/share/usbinit/setupusb reset
+     /bin/bash /system/halium/usr/share/usbinit/setupusb mtp_adb
```
Change paths here as described above to /system/bin/

rubencarneiro replied:
The paths are correct

rubencarneiro committed:
Update Readme

rubencarneiro reviewed (commented):
So you want to be on /system/halium/bin/setupusb?

rubencarneiro committed:
Remove brcm_patchram_plus already prebuilt binaries on averlay

rubencarneiro committed:
Remove mtp-server since is identical to the rootfs

rubencarneiro committed:
Add missing prop to enable mtp

rubencarneiro committed:
Fix handset speaker
removing deepbuffer fixes handset in calls.

rubencarneiro committed:
Update README.md

rubencarneiro committed:
Fix for symlinks missing

rubencarneiro committed:
Fixed camera freeze

rubencarneiro committed:
Fixed video decoding

rubencarneiro committed:
Merge branch 'halium-7.1-updated' of https://github.com/rubencarneiro/android_device_lge_hammerhead into halium-7.1-updated

rubencarneiro committed:
Remove FFMPEG we dont use that

rubencarneiro committed:
Trying to fix timekeeper

rubencarneiro committed:
enable all cpu cores usage

rubencarneiro committed:
Revert cpu cores usage causes instability to device

rubencarneiro committed:
Change default path To MTP Files
Add config file for timekeeper

rubencarneiro committed:
Change script path for MTP

rubencarneiro committed:
Try to also enable adb

rubencarneiro commented:
B

rubencarneiro commented:
Please start the review and merge.

rubencarneiro committed:
Fix typo
Fix missing build option

rubencarneiro committed:
Restore deepbuffer fix audio stuttering during video on morph.
Add flag |AUDIO_OUTPUT_FLAG_PRIMARY to deepbuffer fix in call audio change from earpiece to handset withou removing the deepebuffer.

rubencarneiro committed:
Add Ofono quirks

Flohack74 commented on device.mk:
```diff
@@ -399,6 +421,11 @@ PRODUCT_PROPERTY_OVERRIDES += \
     dalvik.vm.dex2oat-swap=false
 
+# telepathy-ofono quirks
+PRODUCT_PROPERTY_OVERRIDES += \
```
Thats duplicated

rubencarneiro replied:
Fixed

rubencarneiro committed:
Remove duplicate ofono quirks

rubencarneiro committed:
Correct Data time permissions
revert cpu permissions lines changed to end makes no sense
Remove minimediaservice and minisf for now

rubencarneiro committed:
Merge branch 'halium-7.1-updated' of https://github.com/rubencarneiro/android_device_lge_hammerhead into halium-7.1-updated

rubencarneiro committed:
Let try to fix timekeep

rubencarneiro committed:
Add pulse audio config

rubencarneiro committed:
Add acdb_ids

rubencarneiro committed:
symlink wifi/bcmdhd.cal

rubencarneiro committed:
define board gpu

rubencarneiro committed:
remove Gello we dont need that
and other services that we dont need
hciattach and brcm_patchram_plus are rebuilt

rubencarneiro committed:
Try to fix timekeep

rubencarneiro committed:
add otg support

rubencarneiro committed:
remove the amount of rules that we dont need

rubencarneiro committed:
allow tty and property service

rubencarneiro committed:
add more audio permissions

rubencarneiro committed:
Dont need this really

rubencarneiro committed:
Update

rubencarneiro committed:
add minimedia and miniaf services

rubencarneiro committed:
Update rules

rubencarneiro committed:
Remove this that causes a highg usage of cpu and memory

rubencarneiro committed:
Update README.md

UniversalSuperBox commented:
I do not think that this PR can be reviewed any more. @Flohack74 do you agree?

rubencarneiro commented:
> I do not think that this PR can be reviewed any more. @Flohack74 do you agree?

Yes closes all of that.

Flohack74 commented:
@UniversalSuperBox we should however copy over the branch Ruben made to persist that state and history, as I plan to cherry pick a lot of things from there.
